### ISSUE TITLE
Editor crashes after adding or removing components from Actor entity which has orange mesh border displayed in Viewport

### ISSUE THREAD

**Issue by LB-JakubSkorupka** [o3de/o3de#11924]
**Describe the bug**
After adding **any** component to the **Actor** entity with orange border visible in **Viewport** window, Editor crashes.
We noticed that the border around **Actor** entity appears after assigning Actor asset and reselecting mentioned entity, as a result of this, it is still possible to add or remove components from **Actor** entity after initially assigning **Actor** asset in the **Actor** component. 
This issue also occurs on **Linux** OS however it is inconsistent.

The issue occurs with cvar emfx_ragdollManipulatorsEnabled enabled and disabled.

Note that this issue did **not** occur on Development build.

**Assets required**
Rin Actor is included in the AutomatedTesting project.

**Steps to reproduce**
1. Create a new level.
2. Create a new entity.
3. Add Actor and PhysX Ragdoll components to previously created entity.
4. Assign Actor asset in the Actor component.
5. Deselect entity and select it again (so the orange border appear around Actor in the Viewport).
6. Delete Ragdoll component from entity.

**Expected behavior**
Components can be added or removed without issues.

**Actual behavior**
Editor crashes after adding or removing components from the **Actor** entity that have orange border displayed around its mesh in **Viewport**.

**Call Stack Trace:**
```
17) 00007FFB0ED00057 (KERNELBASE) : UnhandledExceptionFilter
16) 00007FFB0E5ED6D8 (InProcessClient64) : (function-name not available)
15) 00007FFB0E5ED321 (InProcessClient64) : (function-name not available)
14) 00007FFB113753B0 (ntdll) : memset
13) 00007FFB1135C766 (ntdll) : _C_specific_handler
12) 00007FFB1137229F (ntdll) : _chkstk
11) 00007FFB11321454 (ntdll) : RtlRaiseException
10) 00007FFB11370DCE (ntdll) : KiUserExceptionDispatcher
 9) C:\build_stabilization\o3de\Gems\Atom\Feature\Common\Code\Source\Mesh\MeshFeatureProcessor.cpp (232) : AZ::Render::MeshFeatureProcessor::GetModel
 8) C:\build_stabilization\o3de\Gems\AtomLyIntegration\EditorModeFeedback\Code\Source\Draw\DrawableMeshEntity.cpp (116) : AZ::Render::DrawableMeshEntity::Draw
 7) C:\build_stabilization\o3de\Gems\AtomLyIntegration\EditorModeFeedback\Code\Source\Draw\EditorStateMaskRenderer.cpp (42) : AZ::Render::EditorStateMaskRenderer::RenderMaskEntities
 6) C:\build_stabilization\o3de\Gems\AtomLyIntegration\EditorModeFeedback\Code\Source\EditorModeFeedbackFeatureProcessor.cpp (127) : AZ::Render::EditorModeFeatureProcessor::Render
 5) C:\build_stabilization\o3de\Code\Framework\AzCore\AzCore\Task\TaskExecutor.cpp (266) : AZ::Internal::TaskWorker::Run
 4) C:\build_stabilization\o3de\Code\Framework\AzCore\Platform\Common\WinAPI\AzCore\std\parallel\internal\thread_WinAPI.cpp (38) : AZStd::Internal::thread_run_function
 3) 00007FFB0F0D1BB2 (ucrtbase) : configthreadlocale
 2) 00007FFB11157034 (KERNEL32) : (function-name not available)
 1) 00007FFB11322651 (ntdll) : RtlUserThreadStart


```

**Video**

https://user-images.githubusercontent.com/86951967/189881092-1510afca-a582-4def-ad68-10822a81083a.mp4

**Found in Branch**
Stabilization/2210 (021d32a)

**Desktop:**
 - Device: PC
 - OS: Windows
 - Version: 10
 - CPU: Intel Core i7-8700K
 - GPU: NVIDIA GeForce RTX 2080 Ti
 - Memory: 16GB

**Additional context**
[error.log](https://github.com/o3de/o3de/files/9556247/error.log)
[error.zip](https://github.com/o3de/o3de/files/9556248/error.zip)
[Editor.log](https://github.com/o3de/o3de/files/9556249/Editor.log)

**Comment by jonawals** [comment]
This issue has been fixed, pending merge.

**Comment by LB-JakubSkorupka** [comment]
Verified the issue as fixed on build Stabilization/2210 (37c7b44)
